common/environment/environment.cpp

```diff
@@ -100,7 +100,7 @@ class CConstDropZoneIteratorByComputer : public CSimpleInterfaceOf<IConstDropZon
 {
 public:
     CConstDropZoneIteratorByComputer(const char * computer);
-    ~CConstDropZoneIteratorByComputer() { if (curr != nullptr) delete curr; };
```

Comment:
delete wasn't correct, but as far as I can see, there needs to be link count handling of curr. It will be leaking an object per first() and next() at the moment.